### PR TITLE
Google output location

### PR DESCRIPTION
**Pull request by jsirianni**
## Description of Changes

Some users would like to se the `location` resource for managed resource types (such as k8s_pod). We do not currently have a way to auto detect this field.

- added `Location` parameter, when not nil, the field will be set to the proto entry's `location` resource field and then deleted.
- make sure `newEntry.Labels` is not nil. If the resource type is not known, this map will be nil and cause a panic when resources are converted to labels [(we do this because google logging will remove any labels that do not map to a monitored resource type](https://github.com/observIQ/stanza/pull/425))
- added test cases for all supported monitored resource types and unknown resource type
- delete resources that were promoted to monitored resource type resources to prevent them from being merged into the entry's labels (duplication)

When an entry fulfills a monitored resource types requirements, the location field is added. For example, `generic_node`:
Screen Shot 2021-09-21 at 1 50 10 PM

input:
```json
{"message":"api connection failed","region":"us-central4","log_type":"custom-app","host.name":"local-cluster"}
{"message":"fail"}
```

config:
```yaml
pipeline:
- type: file_input
  include:
  - ./in.json
  start_at: beginning

- type: json_parser

- type: move
  id: promote-host.name
  from: '$record["host.name"]'
  to: '$resource["host.name"]'

- type: move
  id: promote-log_type
  from: $record.log_type
  to: $labels.log_type

- type: move
  id: promote-region
  from: $record.region
  to: $labels.region
  output:
  - google_cloud_output
  - stdout

- type: stdout

- type: google_cloud_output
  location: $labels.region
  log_name_field: $labels.log_type
  credentials_file: ./creds.json
```

## **Please check that the PR fulfills these requirements**
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] Add a changelog entry (for non-trivial bug fixes / features)
- [x] CI passes

### CODE REVIEW

**Comment by djaglowski** [comment]
| Log Files | Logs / Second | CPU Avg (%) | CPU Avg Δ (%) | Memory Avg (MB) | Memory Avg Δ (MB) |
| --- | --- | --- | --- | --- | --- |
|1|1000|1.4137917|-0.017206669|130.12608|+4.6025085|
|1|5000|5.10351|+0.2587247|133.91582|-2.4260406|
|1|10000|9.207003|-0.74138546|144.11624|-0.3137207|
|1|50000|46.485157|-2.8431015|174.24205|+2.7796326|
|1|100000|82.74281|-11.656128|233.09833|+2.9916534|
|10|100|1.9483917|-0.13792658|133.80658|-0.7421875|
|10|500|5.7242837|-0.03448391|141.73181|+1.2350464|
|10|1000|11.086454|+0.22400475|146.60803|+1.1332245|
|10|5000|54.153442|+2.9974518|181.56412|+2.214981|
|10|10000|107.581856|+10.246269|223.8129|+7.227356|

File: operator/builtin/output/googlecloud/google_cloud.go [hunk 1]
```diff
@@ -52,6 +52,7 @@ type GoogleCloudOutputConfig struct {
 	CredentialsFile     string          `json:"credentials_file,omitempty" yaml:"credentials_file,omitempty"`
 	ProjectID           string          `json:"project_id"                 yaml:"project_id"`
 	LogNameField        *entry.Field    `json:"log_name_field,omitempty"   yaml:"log_name_field,omitempty"`
+	Location            *entry.Field    `json:"location,omitempty"   yaml:"location,omitempty"`
```
Comment (djaglowski) [inline operator/builtin/output/googlecloud/google_cloud.go:55]: Optional: Consider calling this `location_field` to match `log_name_field`.

**Review by djaglowski — DISMISSED** [review]

File: operator/builtin/output/googlecloud/google_cloud.go (hunk 1 above)
Comment (jsirianni) [reply operator/builtin/output/googlecloud/google_cloud.go]: I went back and forth on this, I agree. Done.

**Comment by djaglowski** [comment]
| Log Files | Logs / Second | CPU Avg (%) | CPU Avg Δ (%) | Memory Avg (MB) | Memory Avg Δ (MB) |
| --- | --- | --- | --- | --- | --- |
|1|1000|1.4655426|+0.10341382|126.20286|-0.1627121|
|1|5000|5.3103538|+0.15515375|135.6262|-2.9155579|
|1|10000|9.706845|+0.03448963|143.93265|+0.5078125|
|1|50000|46.3795|+4.7070847|174.86005|+4.586746|
|1|100000|94.07107|+2.667099|225.64413|-17.09671|
|10|100|1.6896782|-0.2758584|133.46376|+0.11920166|
|10|500|5.689665|-0.1035943|137.66649|-2.0207367|
|10|1000|9.89665|-1.8792763|147.49973|-0.33822632|
|10|5000|47.172768|-4.4485397|179.56654|+3.375|
|10|10000|99.67229|+1.0539703|224.80429|+6.9950256|

**Review by djaglowski — APPROVED** [review]

**Pull request merged**